Core/JsRuntime/Source/JsRuntime.cpp

```diff
@@ -0,0 +1,61 @@
+#include "JsRuntime.h"
+
+namespace Babylon
+{
+    namespace
+    {
+        static constexpr auto JS_RUNTIME_NAME = "runtime";
+        static constexpr auto JS_WINDOW_NAME = "window";
+    }
+
+    JsRuntime::JsRuntime(DispatchFunctionT&& dispatchFunction)
```

Comment:
This shouldn't be a &&. There is no reason to force people to move the dispatch function into the constructor.

```suggestion
    JsRuntime::JsRuntime(DispatchFunctionT dispatchFunction)
```

Use std::move like you are doing in the other constructor to move. The calling code can choose whether to move or not.

Reply:
The reasoning here was that this is protected, and that in this case we actually do want to force the function to be moved in. I don't think there's ever a case where we want this function to be passed by copy because there's no reason for anybody else to have this lambda. In such a circumstance, is it better to still leave the choice up to the calling code?

Reply:
Ahh, didn't realize this is protected. Then it doesn't matter that much to me.